pkg/parse/event_handler.go

```diff
@@ -67,22 +67,21 @@ func (s *EventHandler) Handle(event events.Event) events.Result {
 	var runResult RunResult
 	switch event.Type {
 	case events.SyncWithReimportEventType:
-		// Re-apply even if no changes have been detected.
-		// This case should be checked first since it resets the cache.
-		// If the reconciler is in the process of reconciling a given commit, the resync won't
-		// happen until the ongoing reconciliation is done.
-		klog.Infof("It is time for a force-resync")
-		// Reset the cache partially to make sure all the steps of a parse-apply-watch loop will run.
-		// The cached sourceState will not be reset to avoid reading all the source files unnecessarily.
-		// The cached needToRetry will not be reset to avoid resetting the backoff retries.
-		state.resetPartialCache()
-		runResult = runFn(s.Context, s.Reconciler, triggerResync)
+		// Reimport = Read* + Render* + Parse + Update
+		//
+		// * Read & Render will only happen if there's a new commit, new source
+		//   spec change, or a previous error invalidated the cache.
+		//   Otherwise re-import starts from re-parsing the objects from disk.
+		runResult = runFn(s.Context, s.Reconciler, triggerReimport)
```

Comment:
This is the hourly force-resync case. The trigger should be resync.

Reply:
Fixed.